bridged-tokens-wrapper/src/lib.rs

```diff
@@ -0,0 +1,221 @@
+#![no_std]
+
+use transaction::PaymentsVec;
+
+elrond_wasm::imports!();
+
+#[elrond_wasm::contract]
+pub trait BridgedTokensWrapper {
+    #[init]
+    fn init(
+        &self,
+        universal_bridged_token_id: TokenIdentifier,
+        #[var_args] chain_specific_tokens: MultiValueEncoded<TokenIdentifier>,
+    ) {
+        self.universal_bridged_token_id()
+            .insert(universal_bridged_token_id.clone());
+
+        for token_id in chain_specific_tokens {
+            let _ = self
+                .chain_specific_token_ids(&universal_bridged_token_id)
+                .insert(token_id);
+        }
+    }
+
+    #[only_owner]
+    #[endpoint(addWrappedToken)]
+    fn add_wrapped_token(&self, universal_bridged_token_id: TokenIdentifier) {
+        self.universal_bridged_token_id()
+            .insert(universal_bridged_token_id);
+    }
+
+    #[only_owner]
+    #[endpoint(removeWrappedToken)]
+    fn remove_wrapped_token(&self, universal_bridged_token_id: TokenIdentifier) {
+        let _ = self
+            .universal_bridged_token_id()
+            .swap_remove(&universal_bridged_token_id);
+
+        let chain_specific_tokens = &self.chain_specific_token_ids(&universal_bridged_token_id);
+
+        for token in chain_specific_tokens.iter() {
+            let _ = self
+                .chain_specific_token_ids(&universal_bridged_token_id)
+                .swap_remove(&token);
+
+            self.universal_bridged_token_pair(&token).clear();
+        }
+
+        self.universal_bridged_token_id()
```

Comment:
You already removed this at the start of this function.

Reply:
nope.. it is a separate storage.. it should get renamed